updater/cmd/updater/main.go

```diff
@@ -57,100 +47,17 @@ func main() {
-	service := service.NewService(logger, installDir)
-	if err := service.Stop(); err != nil {
-		logger.Error("Failed to stop service", zap.Error(err))
-		fail(logger, installDir)
+	if err := updater.Update(); err != nil {
+		logger.Fatal("Failed to update", zap.Error(err))
 	}
-	// Record that we stopped the service
-	rb.AppendAction(action.NewServiceStopAction(service))
 
-	logger.Debug("Stopped the service")
-
-	if err := rb.Backup(); err != nil {
-		logger.Error("Failed to backup", zap.Error(err))
-
-		// Set the state to failed before rollback so collector knows it failed
-		if setErr := monitor.SetState(packagestate.CollectorPackageName, protobufs.PackageStatus_InstallFailed, err); setErr != nil {
-			logger.Error("Failed to set state on backup failure", zap.Error(setErr))
-		}
-
-		rb.Rollback()
-
-		logger.Error("Rollback complete")
-		fail(logger, installDir)
-	}
-
-	installer := install.NewInstaller(logger, installDir, service)
-	if err := installer.Install(rb); err != nil {
-		logger.Error("Failed to install", zap.Error(err))
-
-		// Set the state to failed before rollback so collector knows it failed
-		if setErr := monitor.SetState(packagestate.CollectorPackageName, protobufs.PackageStatus_InstallFailed, err); setErr != nil {
-			logger.Error("Failed to set state on install failure", zap.Error(setErr))
-		}
-
-		rb.Rollback()
-
-		logger.Error("Rollback complete")
-		fail(logger, installDir)
-	}
-
-	// Create a context with timeout to wait for a success or failed status
-	checkCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-
-	logger.Debug("Installation successful, begin monitor for success")
-
-	// Monitor the install state
-	if err := monitor.MonitorForSuccess(checkCtx, packagestate.CollectorPackageName); err != nil {
-		logger.Error("Failed to install", zap.Error(err))
-
-		// If this is not an error due to the collector setting a failed status we need to set a failed status
-		if !errors.Is(err, state.ErrFailedStatus) {
-			// Set the state to failed before rollback so collector knows it failed
-			if setErr := monitor.SetState(packagestate.CollectorPackageName, protobufs.PackageStatus_InstallFailed, err); setErr != nil {
-				logger.Error("Failed to set state on install failure", zap.Error(setErr))
-			}
-		}
-
-		rb.Rollback()
-
-		logger.Error("Rollback complete")
-		fail(logger, installDir)
-	}
-
-	removeTmpDir(logger, installDir)
-	// Successful update
-	logger.Info("Update Complete")
-}
-
-// fail removes the temporary directory, and calls os.Exit(1)
-func fail(logger *zap.Logger, installDir string) {
-	removeTmpDir(logger, installDir)
-	os.Exit(1)
-}
-
-// removeTmpDir removes the temporary directory and any files in it.
-func removeTmpDir(logger *zap.Logger, installDir string) {
-	err := os.RemoveAll(path.TempDir(installDir))
-	if err != nil {
-		logger.Error("failed to remove temporary directory", zap.Error(err))
-	}
+	logger.Debug("Updater finished successfully")
```

Comment:
This should probably be `Info` level. I don't think much gets logged in the happy case so having some final affirmation always would be good. I know we always pass in debug flag but if we ever change that I don't want us to lose informative logs.